zeebe/engine/src/main/java/io/camunda/zeebe/engine/scaling/redistribution/RedistributionStartedApplier.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.scaling.redistribution;
+
+import io.camunda.zeebe.engine.scaling.redistribution.RedistributionStage.Done;
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.protocol.impl.record.value.scaling.RedistributionProgress;
+import io.camunda.zeebe.protocol.impl.record.value.scaling.RedistributionRecord;
+import io.camunda.zeebe.protocol.record.intent.scaling.RedistributionIntent;
+
+public class RedistributionStartedApplier
+    implements TypedEventApplier<RedistributionIntent, RedistributionRecord> {
+  private final MutableRedistributionState redistributionState;
+
+  public RedistributionStartedApplier(final MutableProcessingState processingState) {
+    redistributionState = processingState.getRedistributionState();
+  }
+
+  @Override
+  public void applyState(final long key, final RedistributionRecord value) {
+    redistributionState.updateState(
+        RedistributionStage.nextStage(new Done()), new RedistributionProgress());
```

Comment:
It's not setting it to `Done`, it's setting it to the "next" stage after `Done`. `Done` is just the stable state meaning that we are not redistributing anything.

Reply:
But the next stage after `Done` is nothing, so it sets it to `Done` instead. If we are not distributing anything, why do we have anything in our state instead of just removing the data when a redistribution is finished?

Reply:
> But the next stage after Done is nothing, so it sets it to Done instead
No, the next stage after Done is `Deployments`. See `RedistributionStageTest`

Reply:
> why do we have anything in our state instead of just removing the data when a redistribution is completed?

We can do that :+1:

Reply:
>No, the next stage after Done is Deployments. See RedistributionStageTest

Ooh you are right, my bad.

Reply:
State is removed now with 916f316d4b3e80cc0f362492969ad239696f7d2f